src/Constraint/MultiConstraint.php

```diff
@@ -203,7 +203,7 @@ public function getUpperBound()
     public static function create(array $constraints, $conjunctive = true)
     {
         if (0 === \count($constraints)) {
-            return new EmptyConstraint();
+            return new MatchAllConstraint();
```

Comment:
I'm wondering whether a MultiConstraint with no constraint provided should actually be a match-all or a match-none (it might have been a case where the name `EmptyConstraint` created confusion).

Reply:
IMO it's fine as is. A Multi with no constraints is like `*` or `""`. It's a lack of constraints, which means everything is matched as it is not constrained. That's why the EmptyConstraint is called like that I believe, it's not that meaningless but it is a bit confusing, for sure MatchAll is more straightforward.